pkg/cloud/websites/websites.go

```diff
@@ -203,6 +222,54 @@ func (l *LocalWebsiteService) Start(websites []Website) error {
 		proxy.ServeHTTP(res, req)
 	})
 
+	// Register the API handler
+	mux.HandleFunc("/ws/{name}", func(w http.ResponseWriter, r *http.Request) {
+		// Get the WebSocket API name from the request path
+		apiName := r.PathValue("name")
+
+		// Get the address of the WebSocket API
+		apiAddress := l.getWebsocketAddress(apiName)
+		if apiAddress == "" {
+			http.Error(w, fmt.Sprintf("WebSocket API %s not found", apiName), http.StatusNotFound)
+			return
+		}
+
+		// Dial the backend WebSocket server
+		targetURL := fmt.Sprintf("ws://%s%s", apiAddress, r.URL.Path)
+		if r.URL.RawQuery != "" {
+			targetURL = fmt.Sprintf("%s?%s", targetURL, r.URL.RawQuery)
+		}
+
+		targetConn, _, err := websocket.DefaultDialer.Dial(targetURL, nil)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to connect to backend WebSocket server: %v", err), http.StatusInternalServerError)
+			return
+		}
+		defer targetConn.Close()
+
+		// Upgrade the HTTP connection to a WebSocket connection
+		upgrader := websocket.Upgrader{}
+
+		clientConn, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to upgrade to WebSocket: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		defer clientConn.Close()
+
+		// Proxy messages between the client and the backend WebSocket server
+		errChan := make(chan error, 2)
+		go proxyWebSocketMessages(clientConn, targetConn, errChan)
+		go proxyWebSocketMessages(targetConn, clientConn, errChan)
+
+		// Wait for an error to occur
+		err = <-errChan
+		if err != nil && !errors.Is(err, websocket.ErrCloseSent) {
+			http.Error(w, fmt.Sprintf("WebSocket proxy error: %v", err), http.StatusInternalServerError)
```

Comment:
Getting an error here due to the connection already being upgraded to a websocket.

```
2025/04/01 14:06:46 http: response.WriteHeader on hijacked connection from github.com/nitrictech/cli/pkg/cloud/websites.(*LocalWebsiteService).Start.func2 (websites.go:269)
                       2025/04/01 14:06:46 http: response.Write on hijacked connection from fmt.Fprintln (print.go:305)
                                                                                                                       2025/04/01 14:06:46 read: websocket: close 1006 (abnormal closure): unexpected EOF
```

Reply:
Have added side logging in the console for this instead and have updated the main websocket error logging as well to make sure it doesn't interrupt the console output